### DISCUSSION TITLE
ROS2 Rust Diagnostics Library

### DISCUSSION THREAD

**Discussion by imdexsam** [ros2-rust/ros2_rust#414] [General]
Does anyone know of an existing implementation of the [ROS2 Diagnostics](https://github.com/ros/diagnostics/tree/ros2) package for Rust?

If not (I'm assuming there isn't) is there a best practice to follow for creating rust implementations of common ROS2 packages?

Obviously there isn't a way to officially make Rust packages for the ROS2 buildfarm etc, but I'm wondering if there are any conventions people follow to help others find and contribute to ros2_rust-targeted packages.

**Comment by romainreignier** [comment]
I am started an implementation here: https://github.com/romainreignier/diagnostic_updater_rs
For now, [`DiagnosedPublisher`](https://github.com/ros/diagnostics/blob/048cd475045e69317309fcecd1d09f7f4e71ea47/diagnostic_updater/include/diagnostic_updater/publisher.hpp#L225C3-L225C21) is missing.
